### ISSUE TITLE
Copy back to local

### ISSUE THREAD

**Issue by carlgrundberg** [deliciousbrains/wp-amazon-s3-and-cloudfront#250]
Did you remove the feature to copy the file back to the local machine?

The filter "as3cf_get_attached_file_copy_back_to_local" doesnt seem to exist in v0.9?

**Comment by polevaultweb** [comment]
The feature still exists, but is implemented differently per plugin that needs the local file using the `as3cf_get_attached_file` filter. How were you using it?

**Comment by carlgrundberg** [comment]
In the previous version (0.8.2) i could just return true in that filter and the s3 plugin downloaded the file.

I used it like this:
```php
// Make sure local file exist when editing image
function skd_as3cf_get_attached_file_copy_back_to_local($copy_back_to_local, $file, $attachment_id) {
    if ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) {
        return $copy_back_to_local;
    }

    if ( isset( $_POST['action'] ) ) { // input var okay
        $action = sanitize_key( $_POST['action'] );
        if($action == 'image-editor' || $action == 'arlima_get_scissors') {
            return true;
        }
    }

    return $copy_back_to_local;
}
add_filter('as3cf_get_attached_file_copy_back_to_local', 'skd_as3cf_get_attached_file_copy_back_to_local', 10, 3);
```

**Comment by polevaultweb** [comment]
Apologies for not keeping that filter.

Take a look at the new plugin compatibility class, you should be able to do something like this filter and callback - https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront/blob/master/classes/as3cf-plugin-compatibility.php#L58

**Comment by carlgrundberg** [comment]
thanks!

**Comment by A5hleyRich** [comment]
Resolved in 0.9.1.
